lib/nodejs/worker.ex

```diff
@@ -50,38 +50,67 @@ defmodule NodeJS.Worker do
     port =
       Port.open(
         {:spawn_executable, node},
-        line: @read_chunk_size,
-        env: [
-          {~c"NODE_PATH", node_path(module_path)},
-          {~c"WRITE_CHUNK_SIZE", String.to_charlist("#{@read_chunk_size}")}
-        ],
-        args: [node_service_path()]
+        [
+          {:line, @read_chunk_size},
+          {:env, get_env_vars(module_path)},
+          {:args, [node_service_path()]},
+          :exit_status,
+          :use_stdio,
+          :stderr_to_stdout
+        ]
       )
 
     {:ok, [node_service_path(), port]}
   end
 
+  defp get_env_vars(module_path) do
+    [
+      {~c"NODE_PATH", node_path(module_path)},
+      {~c"WRITE_CHUNK_SIZE", String.to_charlist("#{@read_chunk_size}")}
+    ]
+  end
+
+  defp maybe_strip_ansi(data) do
+    case data do
+      @prefix ++ _ -> data
+      other -> strip_ansi(other)
+    end
```

Comment:
This part is confusing to me, because we're applying `maybe_strip_ansi` to every chunk, but only the first chunk of the desired function call respose will have the special prefix. Is there something special about that chunk or are we trying to clean all data that's not part of our function call response?

If we're attempting to clean all chunks that aren't part of the function call result, I'd expect the cleaning to come inside the case statement under `:eol`, but then I reach my second point of confusion, because in that clause if `data` doesn't contain the prefix it just gets thrown away anyway. Is it possible you were using `IO.inspect()` or `IO.puts` during testing so that chunks that would normally get thrown away ended up in the terminal?

Reply:
you're right, the main elements seem to be :stderr_to_stdout and properly resetting the terminal output when exiting the process. I was able to completely remove the stripping of the ansi characters and still have the same result